apps/charge-smart-wallets-service/src/smart-wallets/smart-wallets-aa-events.service.ts

```diff
@@ -20,11 +29,41 @@ export class SmartWalletsAAEventsService {
   }
 
   async publishWalletAction (sender, messageData) {
+    if (messageData.name === 'tokenReceive') {
+      this.handleReceiveWalletAction(messageData)
+    }
     try {
       this.centrifugoAPIService.publish(`walletAction:#${sender}`, messageData)
     } catch (error) {
       this.logger.error({ error })
       this.logger.error(`An error occurred during publish message to channel: walletAction:#${sender}`)
     }
   }
+
+  async handleReceiveWalletAction (walletAction) {
+    try {
+      const operator = await callMSFunction(this.accountsClient, 'find-operator-by-smart-wallet', walletAction.walletAddress)
+      if (!operator) {
+        return 'Operator doesnt exist'
+      }
+      const operatorId = operator.ownerId.toString()
+      const user = await callMSFunction(this.accountsClient, 'find-one-user', operatorId)
+      const projectId = (await callMSFunction(this.accountsClient, 'find-one-project-by-owner-id', operatorId))?._id.toString()
+      const apiKey = (await callMSFunction(this.accountsClient, 'get-public', projectId)).publicKey
+      const tokenPriceInUsd = await this.fuseSdkService.getPriceForTokenAddress(walletAction.walletAction.sent[0].address)
+      const amount = formatUnits(walletAction.walletAction.sent[0].value, walletAction.walletAction.sent[0].decimals)
+      const amountUsd = Number(tokenPriceInUsd) * Number(amount)
+      const event = {
+        amount,
+        amountUsd,
+        token: walletAction.sent[0].symbol,
+        apiKey,
+        email: user.email
+      }
+      this.analyticsService.trackEvent('Account Balance Deposited', { ...event }, { user_id: user?.auth0Id })
```

Comment:
@NikolayRn Use the EOA for user id.

Reply:
@LiorAgnin We store operator`s EOA address as authId field on creation, so here im actually using EOA address for userId